src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverage(marks: number[]): number {
+  return marks.reduce(
+    (sum, current) => sum + current, 0,
+  ) / marks.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyOfStudents = [...students];
+  const isAsc: boolean = order === 'asc';
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyOfStudents.sort((a: Student, b: Student) => (isAsc
+        ? a[sortBy].localeCompare(b[sortBy])
+        : b[sortBy].localeCompare(a[sortBy])
+      ));
+      break;
+
+    case SortType.Age:
+      copyOfStudents.sort((a: Student, b: Student) => (isAsc
+        ? a[sortBy] - b[sortBy]
+        : b[sortBy] - a[sortBy]
+      ));
+      break;
+
+    case SortType.Married:
+      copyOfStudents.sort((a: Student, b: Student) => (isAsc
+        ? +a[sortBy] - +b[sortBy]
```

Comment:
```suggestion
        ? Number(a[sortBy]) - Number(b[sortBy])
```
`Number()` is a more obvious way